pkg/skaffold/schema/v1alpha3/config.go

```diff
@@ -0,0 +1,359 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	"fmt"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/constants"
+	kubectx "github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/context"
+	homedir "github.com/mitchellh/go-homedir"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	yaml "gopkg.in/yaml.v2"
+)
+
+const Version string = "skaffold/v1alpha3"
+
+type SkaffoldConfig struct {
+	APIVersion string `yaml:"apiVersion"`
+	Kind       string `yaml:"kind"`
+
+	Build    BuildConfig  `yaml:"build,omitempty"`
+	Deploy   DeployConfig `yaml:"deploy,omitempty"`
+	Profiles []Profile    `yaml:"profiles,omitempty"`
+}
+
+func (c *SkaffoldConfig) GetVersion() string {
+	return c.APIVersion
+}
+
+// BuildConfig contains all the configuration for the build steps
+type BuildConfig struct {
+	Artifacts []*Artifact `yaml:"artifacts,omitempty"`
+	TagPolicy TagPolicy   `yaml:"tagPolicy,omitempty"`
+	BuildType `yaml:",inline"`
+}
+
+// TagPolicy contains all the configuration for the tagging step
+type TagPolicy struct {
+	GitTagger         *GitTagger         `yaml:"gitCommit"`
+	ShaTagger         *ShaTagger         `yaml:"sha256"`
+	EnvTemplateTagger *EnvTemplateTagger `yaml:"envTemplate"`
+	DateTimeTagger    *DateTimeTagger    `yaml:"dateTime"`
+}
+
+// ShaTagger contains the configuration for the SHA tagger.
+type ShaTagger struct{}
+
+// GitTagger contains the configuration for the git tagger.
+type GitTagger struct{}
+
+// EnvTemplateTagger contains the configuration for the envTemplate tagger.
+type EnvTemplateTagger struct {
+	Template string `yaml:"template"`
+}
+
+// DateTimeTagger contains the configuration for the DateTime tagger.
+type DateTimeTagger struct {
+	Format   string `yaml:"format,omitempty"`
+	TimeZone string `yaml:"timezone,omitempty"`
+}
+
+// BuildType contains the specific implementation and parameters needed
+// for the build step. Only one field should be populated.
+type BuildType struct {
+	LocalBuild       *LocalBuild       `yaml:"local"`
+	GoogleCloudBuild *GoogleCloudBuild `yaml:"googleCloudBuild"`
+	KanikoBuild      *KanikoBuild      `yaml:"kaniko"`
+}
+
+// LocalBuild contains the fields needed to do a build on the local docker daemon
+// and optionally push to a repository.
+type LocalBuild struct {
+	SkipPush *bool `yaml:"skipPush"`
+}
+
+// GoogleCloudBuild contains the fields needed to do a remote build on
+// Google Container Builder.
+type GoogleCloudBuild struct {
+	ProjectID string `yaml:"projectId"`
```

Comment:
This part at least has to be synced with master. Might be the case for other parts of the file.